### PR TITLE
Fix for #10474

### PR DESCRIPTION
**Pull request by stereobooster**
If Atom opened with folder e.g. `atom ./` it will save all buffers to workspace tied to this folder.
But if atom opened without folder e.g. `atom` it will prompt if you wan to save all unsaved buffers.
### Testcase
1. Open Atom (without params)
2. Type some text in untitled buffer
3. Quit Atom (Cmd + Q)

Expected behaviour:
1. Shows Prompt whether you want to save or not
2. Depending on user choice close Atom or not 

Current behaviour
1. Close Atom. All text in unsaved buffer is lost

### CODE REVIEW

File: src/text-editor.coffee [hunk 1]
```diff
@@ -875,8 +875,8 @@ class TextEditor extends Model
 
   # Determine whether the user should be prompted to save before closing
   # this editor.
-  shouldPromptToSave: ({windowCloseRequested}={}) ->
-    if windowCloseRequested
+  shouldPromptToSave: ({windowCloseRequested, projectHasPaths}={}) ->
+    if windowCloseRequested && projectHasPaths
```
Comment (thomasjo) [inline src/text-editor.coffee]: Change `&&` to `and`.

**Review by thomasjo — CHANGES_REQUESTED** [review]

File: src/text-editor.coffee (hunk 1 above)
Comment (stereobooster) [inline src/text-editor.coffee]: done

**Comment by stereobooster** [comment]
@thomasjo anything else? Squish commits or something?

**Comment by thomasjo** [comment]
This seems OK to me, but I'll let someone else have a look as well. /cc @atom/feedback

**Comment by iolsen** [comment]
💯 Awesome! Thanks for this contribution!

Merging now and this will be in the 1.13 beta.

**Pull request merged**